syft/src/main/java/org/openmined/syft/data/DataLoader.kt

```diff
@@ -0,0 +1,68 @@
+package org.openmined.syft.data
+
+import org.openmined.syft.data.fetchers.BaseDatasetFetcher
+import org.openmined.syft.data.samplers.BatchSampler
+import org.openmined.syft.data.samplers.RandomSampler
+import org.openmined.syft.data.samplers.Sampler
+import org.openmined.syft.data.samplers.SequentialSampler
+import org.pytorch.IValue
+
+/**
+ * Data loader. Combines a dataset and a sampler, and provides an iterable over
+ * the given dataset. It supports map-style datasets with single-process loading
+ * and customizing loading order.
+ * @param dataset (Dataset)from which to load the data.
+ * @param batchSize (Int, optional): how many samples per batch to load (default: ``1``).
+ * @param shuffle (Boolean, optional) to define the sampler for the dataset
+ * @param dropLast (Boolean, optional): set to ``True`` to drop the last incomplete batch,
+ *                  if the dataset size is not divisible by the batch size. If ``False`` and
+ *                  the size of dataset is not divisible by the batch size, then the last batch
+ *                  will be smaller. (default: ``False``)
+ */
+class DataLoader(var dataset: Dataset,
+                 var batchSize: Int = 1,
+                 var shuffle: Boolean = false,
+                 var dropLast: Boolean = false
+) : Iterable<Pair<IValue, IValue>> {
+
+    private var sampler: Sampler = if (shuffle) {
+        RandomSampler(dataset)
+    } else {
+        SequentialSampler(dataset)
+    }
+
+    private val batchSampler = BatchSampler(
+        sampler,
+        batchSize,
+        dropLast
+    )
+
+    fun indexSampler() = batchSampler
```

Comment:
same as pytorch api

Reply:
Instead of making indexSampler a method, make it an overridable property of the base class then you can just do `override val indexSampler = BatchSampler(sampler,batchSize,dropLast)`